### PR TITLE
Add Integration tests for Transaction Counter

### PR DESCRIPTION
**Pull request by arunans23**
Add Integration tests for Transaction Counter
Resolves https://github.com/wso2/micro-integrator/issues/1639

### CODE REVIEW

File: integration/pom.xml [hunk 1]
```diff
@@ -911,6 +911,15 @@
                 <module>mediation-tests/tests-platform/tests-userstore</module>
             </modules>
         </profile>
+        <profile>
+            <id>transaction-tests</id>
```
Comment (Nirothipan) [inline integration/pom.xml:915]: let's add to db-tests profile as well, to run in Jenkins.

Comment (arunans23) [reply integration/pom.xml:916]: Added.

File: integration/mediation-tests/tests-platform/tests-transaction/src/test/java/org/wso2/micro/integrator/transaction/TransactionCounterUtils.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                pollInterval( POLL_INTERVAL, TimeUnit.SECONDS).
+                atMost(POLL_TIMEOUT, TimeUnit.MINUTES).
+                until(assertTransactionCount(transactionCountResource, requestCount));
+    }
+
+    private static Callable<Boolean> assertTransactionCount(String transactionCountResource, int expectedTransactionCount) {
+        return new Callable<Boolean>() {
+            @Override
+            public Boolean call() throws Exception {
+                return expectedTransactionCount == getTransactionCount(transactionCountResource);
```
Comment (Nirothipan) [inline integration/mediation-tests/tests-platform/tests-transaction/src/test/java/org/wso2/micro/integrator/transaction/TransactionCounterUtils.java:51]: when this fails , can we print the no. of transactions available in db

File: integration/mediation-tests/tests-platform/tests-transaction/pom.xml [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+                <activeByDefault>true</activeByDefault>
+            </activation>
+            <properties>
+                <skipTransactionTests>true</skipTransactionTests>
+            </properties>
+        </profile>
+        <profile>
+            <id>transaction-tests</id>
+            <properties>
+                <skipTransactionTests>false</skipTransactionTests>
```
Comment (Nirothipan) [inline integration/mediation-tests/tests-platform/tests-transaction/pom.xml:288]: need the same to db-tests profile , else will be skipped

File: integration/mediation-tests/tests-platform/tests-transaction/src/test/java/org/wso2/micro/integrator/transaction/TransactionCounterUtils.java (hunk 2 above)
Comment (arunans23) [reply integration/mediation-tests/tests-platform/tests-transaction/src/test/java/org/wso2/micro/integrator/transaction/TransactionCounterUtils.java]: Fixed

File: integration/mediation-tests/tests-platform/tests-transaction/pom.xml (hunk 3 above)
Comment (arunans23) [reply integration/mediation-tests/tests-platform/tests-transaction/pom.xml:288]: Fixed

File: integration/mediation-tests/tests-platform/tests-transaction/pom.xml [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+            </properties>
+        </profile>
+        <profile>
+            <id>db-tests</id>
```
Comment (sdkottegoda) [inline integration/mediation-tests/tests-platform/tests-transaction/pom.xml:292]: what is this profile for? isn't transaction-tests to run tests?

Comment (arunans23) [reply integration/mediation-tests/tests-platform/tests-transaction/pom.xml:292]: We use this profile to run all the platform tests that needs a DB configured (eg: userstore & coordination). https://github.com/wso2/micro-integrator/blob/master/integration/pom.xml#L905

**Pull request merged**